src/components/exercises/App1/redux/slices/contactSlice.js

```diff
@@ -1,54 +1,92 @@
-import {createAsyncThunk, createSlice} from "@reduxjs/toolkit";
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
 import "babel-polyfill";
 
-export const fetchContacts = createAsyncThunk('contacts/fetchContacts', async () => {
-    const response = await fetch('list-contacts.php').then(response => response.json());
+export const fetchContacts = createAsyncThunk(
+  "contacts/fetchContacts",
+  async () => {
+    const response = await fetch("list-contacts.php").then((response) =>
+      response.json()
+    );
     return response.contacts;
-});
+  }
+);
 
 export const contactSlice = createSlice({
-    name: 'contact',
-    initialState: {
-        status: 'idle',
-        data: [],
-        selectedContact: null,
+  name: "contact",
+  initialState: {
+    status: "idle",
+    data: [],
+    selectedContact: null,
+    updateRequired: false,
+  },
+  reducers: {
+    addContact: (state, action) => {
+      state.data.push(action.payload);
+    },
+    updateContact: (state, action) => {
+      const contactId = action.payload;
```

Comment:
- remove unnecessary variables
- use .forEach function to loop contacts
- remove console